pkg/controller/vitesscluster/reconcile_orchestrator.go

```diff
@@ -0,0 +1,183 @@
+		Kind: &corev1.Service{},
+
+		New: func(key client.ObjectKey) runtime.Object {
+			svc := orchestrator.NewService(key, labels)
+			update.ServiceOverrides(svc, vt.Spec.VitessOrchestrator.Service)
+			return svc
+		},
+		UpdateInPlace: func(key client.ObjectKey, obj runtime.Object) {
+			svc := obj.(*corev1.Service)
+			orchestrator.UpdateService(svc, labels)
+			update.InPlaceServiceOverrides(svc, vt.Spec.VitessOrchestrator.Service)
+		},
+		Status: func(key client.ObjectKey, obj runtime.Object) {
+			svc := obj.(*corev1.Service)
+			vt.Status.VitessOrchestrator.ServiceName = svc.Name
+		},
+	})
+	if err != nil {
+		// Record error but continue.
+		resultBuilder.Error(err)
+	}
+
+	// Reconcile orchestrator Deployments.
+	specs := r.orchestratorSpecs(vt, labels)
+
+	// Generate keys (object names) for all desired orchestrator Deployments.
+	// Keep a map back from generated names to the orchestrator specs.
+	keys := make([]client.ObjectKey, 0, len(specs))
+	specMap := make(map[client.ObjectKey]*orchestrator.Spec, len(specs))
+	for _, spec := range specs {
+		key := client.ObjectKey{Namespace: vt.Namespace, Name: orchestrator.DeploymentName(vt.Name, spec.Cell.Name)}
+		keys = append(keys, key)
+		specMap[key] = spec
+	}
+
+	err = r.reconciler.ReconcileObjectSet(ctx, vt, keys, labels, reconciler.Strategy{
+		Kind: &appsv1.Deployment{},
+
+		New: func(key client.ObjectKey) runtime.Object {
+			return orchestrator.NewDeployment(key, specMap[key])
+		},
+		UpdateInPlace: func(key client.ObjectKey, obj runtime.Object) {
+			newObj := obj.(*appsv1.Deployment)
+			if *vt.Spec.UpdateStrategy.Type == planetscalev2.ImmediateVitessClusterUpdateStrategyType {
+				orchestrator.UpdateDeployment(newObj, specMap[key])
+				return
+			}
+			orchestrator.UpdateDeploymentImmediate(newObj, specMap[key])
+		},
+		UpdateRollingInPlace: func(key client.ObjectKey, obj runtime.Object) {
+			newObj := obj.(*appsv1.Deployment)
+			orchestrator.UpdateDeployment(newObj, specMap[key])
+		},
+		Status: func(key client.ObjectKey, obj runtime.Object) {
+			// This function will get called once for each Deployment.
+			// Aggregate as we go to build an overall status for orchestrator.
+			curObj := obj.(*appsv1.Deployment)
+
+			// We'll say orchestrator is Available overall if any of the Deployments is available.
+			// The important thing is that somebody will answer when a client hits the Service.
+			if available := conditions.Deployment(curObj.Status.Conditions, appsv1.DeploymentAvailable); available != nil {
+				// Update the overall status if either we found one that's True, or we previously knew nothing at all (Unknown).
+				if available.Status == corev1.ConditionTrue || vt.Status.VitessOrchestrator.Available == corev1.ConditionUnknown {
+					vt.Status.VitessOrchestrator.Available = available.Status
+				}
+			}
+
+			// TODO(enisoc): Aggregate other important parts of status besides conditions.
```

Comment:
Copy/paste left over?

Reply:
I've left it here because I assume the same TODO applies here also. Orchestrator shares the same concerns that vtctld has.

Reply:
That's a fair point. Ok, I'll resolve this comment.